File: daily-problems/leet-code/1462-course-schedule-iv/main.py
class Solution(object):
    def checkIfPrerequisite(self, numCourses, prerequisites, queries):
        """
        :type numCourses: int
        :type prerequisites: List[List[int]]
        :type queries: List[List[int]]
        :rtype: List[bool]
        """
        reachable = [[False] * numCourses for _ in range(numCourses)]
        
        for u, v in prerequisites:
            reachable[u][v] = True

        for k in range(numCourses):
            for i in range(numCourses):
                for j in range(numCourses):
                    if reachable[i][k] and reachable[k][j]:
                        reachable[i][j] = True
                        
        result = []
        for u, v in queries:
            result.append(reachable[u][v])
        
        return result

File: daily-problems/leet-code/1462-course-schedule-iv/main.go
package main

import "fmt"

func checkIfPrerequisite(numCourses int, prerequisites [][]int, queries [][]int) []bool {
    reachable := make([][]bool, numCourses)
    for i := range reachable {
        reachable[i] = make([]bool, numCourses)
    }

    for _, p := range prerequisites {
        reachable[p[0]][p[1]] = true
    }

    for k := 0; k < numCourses; k++ {
        for i := 0; i < numCourses; i++ {
            for j := 0; j < numCourses; j++ {
                if reachable[i][k] && reachable[k][j] {
                    reachable[i][j] = true
                }
            }
        }
    }

    result := make([]bool, len(queries))
    for i, q := range queries {
        result[i] = reachable[q[0]][q[1]]
    }

    return result
}

func main() {
    numCourses := 3
    prerequisites := [][]int{{1, 2}, {1, 0}, {2, 0}}
    queries := [][]int{{1, 0}, {1, 2}}

    result := checkIfPrerequisite(numCourses, prerequisites, queries)
    fmt.Println(result) 
}
